base/acme/src/main/java/org/dogtagpki/acme/scheduler/ACMEMaintenanceTask.java

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.acme.scheduler;
+
+import java.util.Date;
+
+import org.dogtagpki.acme.database.ACMEDatabase;
+import org.dogtagpki.acme.server.ACMEEngine;
+
+/**
+ * @author Endi S. Dewata
+ */
+public class ACMEMaintenanceTask extends ACMETask {
+
+    public static org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(ACMEMaintenanceTask.class);
+
+    public void run() throws Exception {
+
+        logger.info("Running ACME maintenance");
+
+        Date currentTime = new Date();
+
+        ACMEEngine engine = ACMEEngine.getInstance();
+        ACMEDatabase database = engine.getDatabase();
+
+        database.removeExpiredNonces(currentTime);
```

Comment:
nitpick: One liner?
````
    ACMEEngine.getInstance()
        .getDatabase()
        .remove(ExpiredNonces(currentTime));
````

Reply:
I was planning to add more code like this later:
```
database.removeExpiredAuthorizations(currentTime);
database.removeExpiredOrders(currentTime);
```
so I'm going to leave it as is for now.